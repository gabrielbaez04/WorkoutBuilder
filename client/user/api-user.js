const create = user => fetch('/api/users/', {
  method: 'POST',
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
  body: JSON.stringify(user),
})
  .then(response => response.json()).catch(err => res.status(400).send({
    error: err,
  }));

const list = () => fetch('/api/users/', {
  method: 'GET',
}).then(response => response.json()).catch(err => res.status(400).send({
  error: err,
}));

const read = (params, credentials) => fetch(`/api/users/${params.userId}`, {
  method: 'GET',
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${credentials.t}`,
  },
}).then(response => response.json()).catch(err => res.status(400).send({
  error: err,
}));

const update = (params, credentials, user) => fetch(`/api/users/${params.userId}`, {
  method: 'PUT',
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${credentials.t}`,
  },
  body: JSON.stringify(user),
}).then(response => response.json()).catch(err => res.status(400).send({
  error: err,
}));

const remove = (params, credentials) => fetch(`/api/users/${params.userId}`, {
  method: 'DELETE',
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Bearer ${credentials.t}`,
  },
}).then(response => response.json()).catch(err => res.status(400).send({
  error: err,
}));

export {
  create, list, read, update, remove,
};
