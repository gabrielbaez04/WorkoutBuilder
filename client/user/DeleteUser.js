import React, {Component} from 'react'
import PropTypes from 'prop-types'
import IconButton from '@material-ui/core/IconButton'
import Button from '@material-ui/core/Button'
import DeleteIcon from '@material-ui/icons/Delete'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'
import auth from './../auth/auth-helper'
import {remove} from './api-user.js'
import {Redirect, Link} from 'react-router-dom'

class DeleteUser extends Component {
  state = {
    redirect: false,
    open: false
  }
  clickButton = () => {
    this.setState({open: true})
  }
  deleteAccount = () => {
    this.props.updateMenu();
    const jwt = auth.isAuthenticated()
    remove({
      userId: this.props.userId
    }, {t: jwt.token}).then((data) => {
      if (data.error) {
        console.log(data.error)
      } else {
        auth.signout(() => console.log('deleted'))
        this.props.updateMenu();
        this.setState({redirect: true})
      }
    })
  }
  handleRequestClose = () => {
    this.setState({open: false})
  }
  render() {
    const redirect = this.state.redirect
    if (redirect) {
      return <Redirect to={{pathname: "/", updateMenu: this.props.updateMenu}}/>
    }
    return (<span>
      <IconButton aria-label="Delete" onClick={this.clickButton} color="secondary">
        <DeleteIcon/>
      </IconButton>

      <Dialog open={this.state.open} onClose={this.handleRequestClose}>
        <DialogTitle>{"Delete Account"}</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Confirm to delete your account.
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={this.handleRequestClose} color="primary">
            Cancel
          </Button>
          <Button onClick={this.deleteAccount} color="secondary" autoFocus="autoFocus">
            Confirm
          </Button>
        </DialogActions>
      </Dialog>
    </span>)
  }
}
DeleteUser.propTypes = {
  userId: PropTypes.string.isRequired
}
export default DeleteUser
