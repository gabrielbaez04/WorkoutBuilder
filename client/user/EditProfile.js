import React, { Component } from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import Icon from '@material-ui/core/Icon';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Redirect } from 'react-router-dom';
import auth from '../auth/auth-helper';
import { read, update } from './api-user';

const styles = theme => ({
  card: {
    maxWidth: '95%',
    margin: 'auto',
    textAlign: 'center',
    paddingTop: 5,
    paddingLeft: 5,
    paddingRight: 5,
    marginTop: '24px',
  },
  [theme.breakpoints.up('sm')]: {
    card: {
      maxWidth: '60%',
    },
  },

  title: {
    margin: theme.spacing.unit * 2,
    color: theme.palette.protectedTitle,
  },
  error: {
    verticalAlign: 'middle',
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 300,
  },
  submit: {
    margin: 'auto',
    marginBottom: theme.spacing.unit * 2,
  },
});

class EditProfile extends Component {
  constructor({ match }) {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      redirectToProfile: false,
      error: '',
    };
    this.match = match;
  }

  componentDidMount = () => {
    const jwt = auth.isAuthenticated();
    read({
      userId: this.match.params.userId,
    }, { t: jwt.token }).then((data) => {
      if (data.error) {
        this.setState({ error: data.error });
      } else {
        this.setState({ name: data.name, email: data.email });
      }
    });
  }

  clickSubmit = (e) => {
    const { name, email, password } = this.state;
    e.preventDefault();
    const jwt = auth.isAuthenticated();
    const user = {
      name: name || undefined,
      email: email || undefined,
      password: password || undefined,
    };
    update({
      userId: this.match.params.userId,
    }, {
      t: jwt.token,
    }, user).then((data) => {
      if (data.error) {
        this.setState({ error: data.error });
      } else {
        this.setState({ userId: data._id, redirectToProfile: true });
      }
    });
  }

  handleChange = name => (event) => {
    const { value } = event.target;
    this.setState({ [name]: value });
  }

  render() {
    const { classes } = this.props;
    const {
      userId, name, email, password, error,
    } = this.state;
    if (this.state.redirectToProfile) {
      return (<Redirect to={`/user/${userId}`} />);
    }
    return (
      <form onSubmit={this.clickSubmit}>
        <Card className={classes.card}>
          <CardContent>
            <Typography type="headline" component="h2" className={classes.title}>
              Edit Profile
            </Typography>
            <TextField id="name" label="Name" className={classes.textField} value={name} onChange={this.handleChange('name')} margin="normal" />
            <br />
            <TextField id="email" type="email" label="Email" className={classes.textField} value={email} onChange={this.handleChange('email')} margin="normal" />
            <br />
            <TextField id="password" type="password" label="Password" className={classes.textField} value={password} onChange={this.handleChange('password')} margin="normal" />
            <br />
            {' '}
            {
              error && (
              <Typography component="p" color="error">
                <Icon color="error" className={classes.error}>error</Icon>
                {error}
              </Typography>
              )
            }
          </CardContent>
          <CardActions>
            <Button color="primary" variant="raised" type="submit" className={classes.submit}>Submit</Button>
          </CardActions>
        </Card>
      </form>
    );
  }
}

EditProfile.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(EditProfile);
