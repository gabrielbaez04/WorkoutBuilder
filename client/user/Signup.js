import React, { Component } from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import Icon from '@material-ui/core/Icon';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { Link } from 'react-router-dom';
import { create } from './api-user.js';

const styles = theme => ({
  card: {
    maxWidth: '95%',
    margin: 'auto',
    textAlign: 'center',
    paddingTop: 5,
    paddingLeft: 5,
    paddingRight: 5,
    marginTop: '24px',
  },
  [theme.breakpoints.up('sm')]: {
    card: {
      maxWidth: '60%',
    },
  },
  error: {
    verticalAlign: 'middle',
  },
  title: {
    marginTop: theme.spacing.unit * 2,
    color: theme.palette.openTitle,
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 300,
  },
  submit: {
    margin: 'auto',
    marginBottom: theme.spacing.unit * 2,
  },
});

class Signup extends Component {
  state = {
    name: '',
    password: '',
    email: '',
    open: false,
    error: '',
  }

  handleChange = name => (event) => {
    this.setState({ [name]: event.target.value });
  }

  clickSubmit = (e) => {
    e.preventDefault();
    const user = {
      name: this.state.name || undefined,
      email: this.state.email || undefined,
      password: this.state.password || undefined,
    };
    create(user).then((data) => {
      if (data.error) {
        this.setState({ error: data.error });
      } else {
        this.setState({ error: '', open: true });
      }
    });
  }

  render() {
    const { classes } = this.props;
    return (
      <div>
        <form onSubmit={this.clickSubmit}>
          <Card className={classes.card}>
            <CardContent>
              <Typography type="headline" component="h2" className={classes.title}>
                Sign Up
              </Typography>
              <TextField id="name" label="Name" className={classes.textField} value={this.state.name} onChange={this.handleChange('name')} margin="normal" />
              <br />
              <TextField id="email" type="email" label="Email" className={classes.textField} value={this.state.email} onChange={this.handleChange('email')} margin="normal" />
              <br />
              <TextField id="password" type="password" label="Password" className={classes.textField} value={this.state.password} onChange={this.handleChange('password')} margin="normal" />
              <br />
              {' '}
              {
                this.state.error && (
                <Typography component="p" color="error">
                  <Icon color="error" className={classes.error}>error</Icon>
                  {this.state.error}
                </Typography>
                )
              }
            </CardContent>
            <CardActions>
              <Button color="primary" variant="raised" type="submit" className={classes.submit}>Submit</Button>
            </CardActions>
          </Card>
        </form>
        <Dialog open={this.state.open} disableBackdropClick>
          <DialogTitle>New Account</DialogTitle>
          <DialogContent>
            <DialogContentText>
              New account successfully created.
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Link to={{ pathname: '/signin', updateMenu: this.props.location.updateMenu }}>
              <Button color="primary" autoFocus="autoFocus" variant="raised">
                Sign In
              </Button>
            </Link>
          </DialogActions>
        </Dialog>
      </div>);
  }
}

Signup.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Signup);
