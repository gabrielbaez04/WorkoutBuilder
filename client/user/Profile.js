import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Edit from '@material-ui/icons/Edit';
import Person from '@material-ui/icons/Person';
import Divider from '@material-ui/core/Divider';
import { Redirect, Link } from 'react-router-dom';
import DeleteUser from './DeleteUser';
import auth from '../auth/auth-helper';
import { read } from './api-user.js';

const styles = theme => ({
  root: theme.mixins.gutters({
    maxWidth: '95%',
    margin: 'auto',
    paddingTop: 5,
    paddingLeft: 5,
    paddingRight: 5,
    marginTop: '24px',
    [theme.breakpoints.up('sm')]: {
      maxWidth: '60%',
    },
  }),
  title: {
    margin: `${theme.spacing.unit * 3}px 0 ${theme.spacing.unit * 2}px`,
    color: theme.palette.protectedTitle,
  },
});

class Profile extends Component {
  constructor({ match }) {
    super();
    this.state = {
      user: '',
      redirectToSignin: false,
    };
    this.match = match;
  }

  init = (userId) => {
    const jwt = auth.isAuthenticated();
    read({
      userId,
    }, { t: jwt.token }).then((data) => {
      if (data.error) {
        this.setState({ redirectToSignin: true });
      } else {
        this.setState({ user: data });
      }
    });
  }

  componentWillReceiveProps = (props) => {
    this.init(props.match.params.userId);
  }

  componentDidMount = () => {
    this.init(this.match.params.userId);
  }

  render() {
    const { classes } = this.props;
    const redirectToSignin = this.state.redirectToSignin;
    if (redirectToSignin) {
      return <Redirect to={{ pathname: '/signin', updateMenu: this.props.location.updateMenu }} />;
    }
    return (
      <Paper className={classes.root} elevation={4}>
        <Typography type="title" className={classes.title}>
          Profile
        </Typography>
        <List dense>
          <ListItem>
            <ListItemAvatar>
              <Avatar>
                <Person />
              </Avatar>
            </ListItemAvatar>
            <ListItemText primary={this.state.user.name} secondary={this.state.user.email} />
            {' '}
            {
             auth.isAuthenticated().user && auth.isAuthenticated().user._id == this.state.user._id
              && (
              <ListItemSecondaryAction>
                <Link to={`/user/edit/${this.state.user._id}`}>
                  <IconButton aria-label="Edit" color="primary">
                    <Edit />
                  </IconButton>
                </Link>
                <DeleteUser userId={this.state.user._id} updateMenu={this.props.location.updateMenu} />
              </ListItemSecondaryAction>
              )
            }
          </ListItem>
          <Divider />
          <ListItem>
            <ListItemText primary={`Joined: ${(
              new Date(this.state.user.created)).toDateString()}`}
            />
          </ListItem>
        </List>
      </Paper>
    );
  }
}
Profile.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Profile);
