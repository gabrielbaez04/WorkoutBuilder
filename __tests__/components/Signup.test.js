import React from 'react';
import { shallow } from 'enzyme';
import { Signup } from '../../client/user/Signup';
import "isomorphic-fetch"

let wrapper, updateMenu;

beforeEach(() => {
    updateMenu = jest.fn();
    wrapper = shallow(<Signup classes={{}} location={{ updateMenu }} />);
});

test('should render Signup correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should set name on name change', () => {
    const value = 'Gabriel';
    wrapper.find('#name').prop('onChange')({ target: { value } });
    expect(wrapper.state('name')).toBe(value);
});

test('should set email on email change', () => {
    const value = 'Gabrielbaez04@gmail.com';
    wrapper.find('#email').prop('onChange')({ target: { value } });
    expect(wrapper.state('email')).toBe(value);
});

test('should set password on password change', () => {
    const value = 'gabriel123';
    wrapper.find('#password').prop('onChange')({ target: { value } });
    expect(wrapper.state('password')).toBe(value);
});
