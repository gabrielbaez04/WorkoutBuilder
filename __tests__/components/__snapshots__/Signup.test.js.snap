// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Signup correctly 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <WithStyles(Card)>
      <WithStyles(CardContent)>
        <WithStyles(Typography)
          component="h2"
          type="headline"
        >
          Sign Up
        </WithStyles(Typography)>
        <TextField
          id="name"
          label="Name"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
          variant="standard"
        />
        <br />
        <TextField
          id="email"
          label="Email"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          type="email"
          value=""
          variant="standard"
        />
        <br />
        <TextField
          id="password"
          label="Password"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          type="password"
          value=""
          variant="standard"
        />
        <br />
         
      </WithStyles(CardContent)>
      <WithStyles(CardActions)>
        <WithStyles(Button)
          color="primary"
          type="submit"
          variant="raised"
        >
          Submit
        </WithStyles(Button)>
      </WithStyles(CardActions)>
    </WithStyles(Card)>
  </form>
  <WithStyles(Dialog)
    disableBackdropClick={true}
    open={false}
  >
    <WithStyles(DialogTitle)>
      New Account
    </WithStyles(DialogTitle)>
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)>
        New account successfully created.
      </WithStyles(DialogContentText)>
    </WithStyles(DialogContent)>
    <WithStyles(DialogActions)>
      <Link
        replace={false}
        to={
          Object {
            "pathname": "/signin",
            "updateMenu": [MockFunction],
          }
        }
      >
        <WithStyles(Button)
          autoFocus="autoFocus"
          color="primary"
          variant="raised"
        >
          Sign In
        </WithStyles(Button)>
      </Link>
    </WithStyles(DialogActions)>
  </WithStyles(Dialog)>
</div>
`;
